pandas/_libs/parsers.pyx

```diff
@@ -927,9 +927,12 @@ cdef class TextReader:
                 (num_cols >= self.parser.line_fields[i]) * num_cols
 
         if self.table_width - self.leading_cols > num_cols:
-            raise ParserError(f"Too many columns specified: expected "
-                              f"{self.table_width - self.leading_cols} "
-                              f"and found {num_cols}")
+            if ((not callable(self.usecols) and self.usecols and
```

Comment:
can you split these conditions up a bit (e.g. use temp variables) to make this easier to read. a

Reply:
is this better?